### DISCUSSION TITLE
saspy with Reticulate in Rstudio

### DISCUSSION THREAD

**Discussion by nivelw** [sassoftware/saspy#483] [Q&A] [answered]
Hi, 

I have been using saspy for a few months within VSCode now and it has been a good experience. Recently, I am running some R scripts and wanted to run .py files in my project using the `reticulate` package in R. There isn't any issue until I tried running saspy within `reticulate`. Specifically, the R session seemed hang after I've keyed in user id at `saspy.SASsession()` prompt. After that the cursor moved to next line and keeps blinking as if it is waiting for something else to load. 

Posting here hoping to get insights from saspy perspective. Below is the python command I ran via `reticulate`. 

```
>>> import pandas as pd
>>> import saspy
>>> from pathlib import Path
>>> sas=saspy.SASsession()
Please enter the IOM user id: 
```

Thanks.

**Comment by nivelw** [comment, marked as answer]
Managed to find two ways for this. One is to include `omruser` and `omrpw` in `SASsession()`. This would mean exposing userid and password thus not recommended. The other, which is more more secured, is to use a _authinfo file (or .authinfo in linux). Reference to user guide below.

https://sassoftware.github.io/saspy/configuration.html#the-authinfo-file-authinfo-on-windows
https://sassoftware.github.io/saspy/configuration.html#remote

**Comment by tomweber-sas** [comment]
So, when using vscode, you were typing in your credentials every time you connected? You hadn't put them in a authinfo file or in your config file? Sounds like something in your R setup doesn't support prompting in that environment; if you can't type in when prompted. So yes, using the authinfo file is the best way in both of these environments; rather than being prompted in the first place; where you actually can type in a response to the prompt, and certainly where it doesn't support prompting correctly!
